### PR TITLE
fix bug Painting bug in IconEditor

### PR DESCRIPTION
**Pull request by Epica3055**
Fix #[5092](https://github.com/dotnet/winforms/issues/5092)

## Regression? 

-  No

## Risk

- Minimal

### Before

The icon property of the NotifyIcon control cannot be modified

image

### After
The icon property of the NotifyIcon control can be modified

image

## Test methodology 

- Manually (Verified in Scratch project)

## Test environment(s) 

- .net 9.0.0-alpha.1.23510.12

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10131)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Drawing/Design/IconEditor.cs [hunk 1]
```diff
@@ -119,12 +119,14 @@ public override void PaintValue(PaintValueEventArgs e)
         if (icon.Width < rectangle.Width)
         {
             rectangle.X = (rectangle.Width - icon.Width) / 2;
+            rectangle.X += (rectangle.Width - icon.Width) / 2;
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/Drawing/Design/IconEditor.cs:122]: Surely both these can now be consolidated to `rectangle.X = rectangle.Width - icon.Width;`

Comment (Epica3055) [reply src/System.Windows.Forms.Design/src/System/Drawing/Design/IconEditor.cs:122]: I am sorry, what did you mean?

Comment (hughbe) [reply src/System.Windows.Forms.Design/src/System/Drawing/Design/IconEditor.cs:122]: We've assigned `X` to be `(rectangle.Width - icon.Width) / 2`, then we add `(rectangle.Width - icon.Width) / 2` to it.

So we can just assign `X` to be `(rectangle.Width - icon.Width)`

Comment (Epica3055) [reply src/System.Windows.Forms.Design/src/System/Drawing/Design/IconEditor.cs:122]: Sorry for the confusing code.  I must not have enough coffee the other day.  😴 
``` c#
    public override void PaintValue(PaintValueEventArgs e)
    {
        if (e?.Value is not Icon icon)
        {
            return;
        }
        // If icon is smaller than rectangle, just center it unscaled in the rectangle.
        Rectangle rectangle = e.Bounds;
        if (icon.Width < rectangle.Width)
        {
            rectangle.X = (rectangle.Width - icon.Width) / 2;
            rectangle.Width = icon.Width;
        }

        if (icon.Height < rectangle.Height)
        {
            rectangle.X = (rectangle.Height - icon.Height) / 2;
            rectangle.Height = icon.Height;
        }

        e.Graphics.DrawIcon(icon, rectangle);
    }
```
As we can see, this method is to re-calculate the icon position.

The logic here is not right
``` c#
        if (icon.Width < rectangle.Width)
        {
            rectangle.X = (rectangle.Width - icon.Width) / 2;
            rectangle.Width = icon.Width;
        }
```
As we can see:  `(rectangle.Width - icon.Width) / 2` is the offsetX. I think it is not right to assign offsetX to `rectangle.X`, which I think is the cause of this issue.
To address this error is simple, just `rectangle.X = rectangle.X + (rectangle.Height - icon.Height) / 2`, which is `rectangle.X += (rectangle.Width - icon.Width) / 2;`.
But for some reason, I forgot to remove the old code.

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, but please double check that runtime property grid scenario had not regressed, in the case of icons smaller than 16x16. Also consider adding an Icon property to https://github.com/dotnet/winforms/blob/573d21d5ca575b940186f5a1a29512a28a682185/src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/UserControls/UserControlWithObjectCollectionEditor.cs#L11 , and adding a smaller icon resource to the project for easy experiments

**Comment by Tanya-Solyanik** [comment]
Never mind, I noticed that icon drawing in property grid is blocked by an unexpected transform - https://github.com/dotnet/winforms/pull/10131

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**